core/src/test/java/hudson/util/io/ZipArchiverTest.java

```diff
@@ -2,173 +2,84 @@
-        }
-    }
-
-    @After
-    public void tearDown() {
-        deleteDir(tmpDir);
-    }
+    @Rule public TemporaryFolder tmp = new TemporaryFolder();
 
     @Issue("JENKINS-9942")
     @Test
-    public void backwardsSlashesOnWindows()  {
+    public void backwardsSlashesOnWindows() throws IOException {
         // create foo/bar/baz/Test.txt
-        File tmpFile = null;
-        try {
-            File baz = new File(new File(new File(tmpDir, "foo"), "bar"), "baz");
-            baz.mkdirs();
-
-            tmpFile = new File(baz, "Test.txt");
-            tmpFile.createNewFile();
-        } catch (IOException e) {
-            fail("unable to prepare source directory for zipping", e);
-        }
+        Path baz = tmp.newFolder().toPath().resolve("foo").resolve("bar").resolve("baz");
+        Files.createDirectories(baz);
+        Path tmpFile = baz.resolve("Test.txt");
+        Files.createFile(tmpFile);
 
         // a file to store the zip archive in
-        File zipFile = null;
+        Path zipFile = Files.createTempFile(tmp.getRoot().toPath(), "test", ".zip");
 
         // create zip from tmpDir
-        ZipArchiver archiver = null;
-
-        try {
-            zipFile = File.createTempFile("test", ".zip");
-            archiver = new ZipArchiver(Files.newOutputStream(zipFile.toPath()));
-
-            archiver.visit(tmpFile, "foo\\bar\\baz\\Test.txt");
-        } catch (Exception e) {
-            fail("exception driving ZipArchiver", e);
-        } finally {
-            if (archiver != null) {
-                try {
-                    archiver.close();
-                } catch (IOException e) {
-                    // ignored
-                }
-            }
+        try (ZipArchiver archiver = new ZipArchiver(Files.newOutputStream(zipFile))) {
+            archiver.visit(tmpFile.toFile(), "foo\\bar\\baz\\Test.txt");
         }
 
         // examine zip contents and assert that none of the entry names (paths) have
         // back-slashes ("\")
-        String zipEntryName = null;
-
-        try (ZipFile zipFileVerify = new ZipFile(zipFile)) {
-
-            zipEntryName = ((ZipEntry) zipFileVerify.entries().nextElement()).getName();
-        } catch (Exception e) {
-            fail("failure enumerating zip entries", e);
+        try (ZipFile zipFileVerify = new ZipFile(zipFile.toFile())) {
+            assertEquals(1, zipFileVerify.size());
+            ZipEntry zipEntry = zipFileVerify.entries().nextElement();
+            assertEquals("foo/bar/baz/Test.txt", zipEntry.getName());
         }
-
-        assertEquals("foo/bar/baz/Test.txt", zipEntryName);
     }
 
     @Test
-    public void huge64bitFile()  {
+    public void huge64bitFile() throws IOException {
         // create huge64bitFileTest.txt
-
-        File hugeFile = new File(tmpDir, "huge64bitFileTest.txt");
+        Path hugeFile = tmp.newFolder().toPath().resolve("huge64bitFileTest.txt");
+        long length = 4L * 1024 * 1024 * 1024 + 2;
```

Comment:
The key fix to the test: ensuring that the length doesn't overflow by using "4L" instead of "4".